### ISSUE TITLE
Discrepancy in nodetool output and tlp-reaper UI Screen:

### ISSUE THREAD

**Issue by AsadAliZaidi** [thelastpickle/cassandra-reaper#332]
why there is discrepancy in nodetool output and in tlp-reaper UI Screen??

**Environment:** apache Cassandra 3.11.1 – multi (two) datacenter setup.
**Tlp-reaper:** 1.0.2

I see there is difference in reaper report and nodetool output,  screenshot is attached – you can see nodetool is reporting 256 token for node hosted at (216.X.X.14) server with 567.12 GB of load but reaper says tokens are only _20_ with _609GB_ on disk?.  

Wondering why there is discrepancy in numbers?

**Nodetool output:**
 
![nodetool_output](https://user-images.githubusercontent.com/7247196/35992325-978c34b4-0ccf-11e8-83a5-b6cb604a0db8.png)

**Tlp-reaper report:**
 
![tlp_reaper_reporting](https://user-images.githubusercontent.com/7247196/35992357-b00d1b52-0ccf-11e8-92c6-2edbc14eec8e.png)

**Comment by adejanovski** [comment]
Reaper uses the values that you can get from `nodetool gossipinfo` and apparently the number of tokens is broken there (I can't speak for data size on disk).
I'll file a ticket in the Cassandra JIRA to fix the number of tokens, as it's been broken for ages.

I could try to fix the number of tokens by querying the token to endpoint map and compute it from there.

I'll update the issue once I have a proper fix.

**Comment by a1exsh** [comment]
@adejanovski could you please refer to that JIRA here?  We've just seen a really weird tokens number on one of our nodes:
![image](https://user-images.githubusercontent.com/489601/47069808-294bf480-d1f0-11e8-951d-1c026d9411e3.png)

We are using 16 tokens per node and the rest report something between 13 and 15, which is also weird, but this number is too weird.  I can confirm it comes from Cassandra itself:
```
$ docker exec taupageapp nodetool gossipinfo | grep TOKENS
  TOKENS:13:<hidden>
  TOKENS:128817:<hidden>
  TOKENS:13:<hidden>
...
```

**Comment by adejanovski** [comment]
It's actually our fault here, the number that we see isn't related to the number of tokens but is a version number : https://github.com/apache/cassandra/blob/cassandra-2.2/src/java/org/apache/cassandra/gms/FailureDetector.java#L204

The number of tokens is not accessible through the failure detector. 
We need to retrieve it somewhere else and in the meantime we'll probably remove this from the UI.

**Comment by a1exsh** [comment]
Argh.  OK.  I wonder now what could explain so wildly different token version in our case.  Any idea what those are?

**Comment by adejanovski** [comment]
Fixed in master by #582
